pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -281,7 +281,7 @@ in package-set { inherit pkgs lib callPackage; } self // {
     # GHC is setup with a package database with all the specified Haskell packages.
     #
     # ghcWithPackages :: (HaskellPkgSet -> [ HaskellPkg ]) -> Derivation
-    ghcWithPackages = self.callPackage ./with-packages-wrapper.nix {
```

Comment:
I don't think I have enough knowledge about cross-compiling to be able to judge this PR.  But I do have a couple questions that will hopefully be helpful for my understanding:

1.  I'm guessing that some of the arguments to https://github.com/NixOS/nixpkgs/blob/c66d7f97164b608972a15bc0936fce9a53bf4c66/pkgs/development/haskell-modules/with-packages-wrapper.nix specifically need to come from `buildHaskellPackages` instead of `self`.  Which arguments?  `stdenv` and `hoogleWithPackages`?

2. Can you give an example of how you'd call `ghcWithPackages` in some way where it would break when using `self.callPackage` (as currently in `master`), but not when using `buildHaskellPackages.callPackage` (as in this PR)?

Reply:
The mistake is most apparent for `pkgsCross.ghcjs.haskellPackages.ghcWithPackages` were it'd try to build `bash` for `javascript-unknown-ghcjs`. This doesn't really work as a test though, since such a derivation won't finish building even after this change due to the `ranlib` problem on that platform.

A test would be an execution test of a cross-compiled ghc with packages environment, *but* since it relies on shell wrappers and qemu user emulation can't intercept `fork()` it is not really possible to write a good test for this at the moment, I think.

Reply:
That makes sense, thanks for the explanation!